jvm-packages/xgboost4j-spark/src/test/scala/ml/dmlc/xgboost4j/scala/spark/XGBoostGeneralSuite.scala

```diff
@@ -235,4 +233,30 @@ class XGBoostGeneralSuite extends FunSuite with PerTest {
     assert(error(prevModel._booster) > error(nextModel._booster))
     assert(error(nextModel._booster) < 0.1)
   }
+
+  test("LabeledPointGroupIterator with group data") {
+    val groups = new LabeledPointGroupIterator(Ranking.train.iterator).toArray
+    assert(groups.length == 20)
+    // first group
+    assert(groups(0)._1)
+    assert(groups(0)._2(0).group == 1)
+    // last group
+    assert(groups(19)._1)
+    assert(groups(19)._2(0).group == 20)
+    // normal group
+    assert(!groups(15)._1)
+    assert(groups(15)._2(0).group == 16)
+  }
+
+  test("distributed training with group data") {
```

Comment:
is there any making-sense but small dataset we can use for ranking unit test? I think the current unit test regarding ranking are all testing *if the training can finished*

Reply:
This PR is to make sure that, after training data is repartitioning and splitted, the training groups are still correct and the order of LabeledPoints within a group is kept.
Yes, it only tests if training can finished. I can not use my training data from Apple, so let's see if we could find some sample data set for ranking.